src/fixpoint/core.clj

```diff
@@ -158,18 +158,21 @@
 ;; ## Fixture Insertion
 
 (defn- lookup-reference
-  [entities document-id path]
+  [entities document-id transformations]
   (assert
     (contains? entities document-id)
     (str "no such document available within the current fixture scope: "
          (pr-str document-id)))
-  (let [result (get-in entities (cons document-id path) ::none)]
-    (assert (not= result ::none)
-            (format "document '%s' does not contain property '%s': %s"
-                    (pr-str document-id)
-                    (pr-str (vec path))
-                    (pr-str (get entities document-id))))
-    result))
+  (if (every? keyword? transformations)
+    (let [result (get-in entities (cons document-id transformations) ::none)]
+      (assert (not= result ::none)
+              (format "document '%s' does not contain property '%s': %s"
+                      (pr-str document-id)
+                      (pr-str (vec transformations))
+                      (pr-str (get entities document-id))))
+      result)
+    (let [document (get entities document-id)]
+      (reduce #(%2 %1) document transformations))))
```

Comment:
It'd be nicer to use `fn` here and give `%2` and `%1` a bit more descriptive names.